ONSdigital/dp-frontend-dataset-controller
# Feature/get filtered output

### What

Created get `FilterOutput` handler, as this handler creates a `census landing page` a lot of the handler calls are inherited from the `censusLanding` handler with additional calls to `getAreas` from the dimensions client and `getOutput` from the filter client.
There is future work to make the API calls concurrent.

Resolves frontend tickets [5633](https://trello.com/c/srqSjE6n/5633-handle-filter-output-route-in-dp-frontend-dataset-controller), [5634](https://trello.com/c/BFL0Zk2Q/5634-alter-variables-table-to-display-dimensions-selected-by-user) and [5635](https://trello.com/c/CFMIo8FC/5635-retrieve-downloads-from-filter-outputs-get-endpoint) as part of user story [5401: Change geography Area Type - nonJS](https://trello.com/c/uMRrZHB5/5401-5401-change-geography-area-type-nonjs)

### How to review

- Sense check
- Tests pass
- Image review
- _Optionally_ run the stack locally; using the [cantabular-import](https://github.com/ONSdigital/dp-compose/tree/main/cantabular-import) docker container. 
  - Create a flexible dataset
  - Go to a [dataset landing page](http://localhost:8081/datasets/cantabular-flexible-example/editions/2021/versions/1)
  - Click 'change' 
  - Change the geography
  - Check the summary screen has your changed geography
  - Click 'get the data' button
  - You should be redirected to a [filtered dataset landing page](http://localhost:8081/datasets/cantabular-flexible-example/editions/2021/versions/1/filter-outputs/fbe9204c-b1e0-435a-a177-603677ee187d)
  - _Note_ downloads are not working locally!
  - **OR** call me 🤙 and I'll show you 

#### Image
**Start of journey**

<img width="1151" alt="flexible dataset landing page" src="https://user-images.githubusercontent.com/19624419/169509172-7e249a94-dd10-48e6-a7d9-30331ba44cc4.png">

**Changed geography**
_Note_ different route and changed dimensions

<img width="1151" alt="filtered output" src="https://user-images.githubusercontent.com/19624419/169508809-01572d18-9e9b-4c17-a6c4-fb88c2050d76.png">

### Who can review

Frontend go dev
